### ISSUE TITLE
Friendlier error messages when file not found

### ISSUE THREAD

**Issue by niclasmattsson** [evetion/GeoDataFrames.jl#49]
This is an upstream problem in ArchGDAL but I'm opening it here (since this package seems to be aimed at making ArchGDAL's weird pointer-based types friendlier to work with). Here's the problem in a nutshell:

```julia
julia> gdf = GDF.read("doesnt_exist.gpkg")
ERROR: GDALError (CE_Failure, code 10):
        Pointer 'hDS' is NULL in 'GDALDatasetH'.
```

Could we have `read()` check if a file exists before passing it on to ArchGDAL, and throw a friendlier error message if it doesn't exist?

**Comment by visr** [comment]
I agree this is not a great error message. It comes directly from GDAL, and we probably catch it in GDAL.jl using out custom error handler: https://github.com/JuliaGeo/GDAL.jl/blob/v1.5.1/src/error.jl#L49-L63

Doing an `isfile(path)` before handing it to GDAL prevents us from using URLs or [virtual file systems](https://gdal.org/user/virtual_file_systems.html) however, so I don't see that working.

Only thing I can think of is to perhaps avoid the GDAL.jl `aftercare` function called in https://github.com/yeesian/ArchGDAL.jl/blob/v0.10.1/src/dataset.jl#L536 (and similar), and instead directly [use ccall](https://github.com/JuliaGeo/GDAL.jl/blob/v1.5.1/src/libgdal.jl#L6403-L6412). That would have to be done in ArchGDAL.jl

**Comment by evetion** [comment]
Do you have the latest version installed? I fixed that in v`0.3.2`, some months back.

```
julia> gdf = GeoDataFrames.read("doesnt_exist.gpkg")
ERROR: File not found.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] read(fn::String; kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ GeoDataFrames ~/.julia/packages/GeoDataFrames/pHXgK/src/io.jl:32
 [3] read(fn::String)
   @ GeoDataFrames ~/.julia/packages/GeoDataFrames/pHXgK/src/io.jl:31
 [4] top-level scope
   @ REPL[4]:1
```

**Comment by alex-s-gardner** [comment]
Can this be closed?
